feat: contribution guide

Dudleyneedham opened the pull request:
## fixes https://github.com/KILTprotocol/ticket/issues/1847

Adds the contribution

## How to test:

yarn start

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

Dudleyneedham committed:
feat: adding contribution

Dudleyneedham committed:
feat: adding more details

tjwelde commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,45 @@
+ - Open an issue on the corresponding repository
+ - Provide a written outline of the feature request for discussion
+
+At the end of the discussion, the proposal needs the approval of the community, KILT Technical Committee, governance, or relevant project team (depending on the subject) to progress to the next stage. If approved, acceptable deliverables will be agreed with the proposer.
```
So is this just for requesting features that KILT develops, or is it proposing a feature, that the proposer develops?

Dudleyneedham replied:
I guess this could be both.

There could be a final closing remark here stating that the proposal or request must be taken by a team or done by the proposer.

tjwelde replied:
I like the split, but I think in the "Feature Proposal" should mention, that a proposal has to be made. because there is no proposal at that point.

Dudleyneedham committed:
feat: separating feaature requests and proposals

ntn-x2 commented on docs/development/00-introduction.md:
```diff
@@ -0,0 +1,8 @@
+title: Development
+---
+
+KILT protocol is an open-source project built upon the [substrate framework](https://docs.substrate.io/). The guide is to help builders and contributors find the resources needed to take action and work within the guideance of the active developers.
```
```suggestion
KILT protocol is an open-source project built upon the [Substrate framework](https://docs.substrate.io/). The guide is to help builders and contributors find the resources needed to take action and work within the guidance of the active developers.
```

ntn-x2 commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,46 @@
+
+For the integrity of the protocol, requesting a feature requires approval from KILT governance. The request should follow the guidelines below, which are subject to change:
+
+ - Begin a discussion with the community to ensure that they agree that it adds to the KILT protocol in a meaningful way, supporting its goals
```
Maybe?

```suggestion
 - Begin a discussion with the community to ensure that they agree that it adds value to the KILT protocol in a meaningful way, supporting its goals
```

Dudleyneedham committed:
feat: Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

Dudleyneedham committed:
feat: finishing touches

Dudleyneedham committed:
feat: adding a line

tjwelde commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,46 @@
+
+## Feature Proposals
+
+A proposal must be made before moving ahead, see above. At the end of the discussion, the proposal needs the approval of the community, KILT Technical Committee, governance, or relevant project team (depending on the subject) to progress to the next stage. If approved, acceptable deliverables will be agreed with the proposer, and the active developers or appropriate teams can start working on the ...
```
I still think it is confusing and mixing things up. AFAICS, this section should just be about proposal, which the proposer will have to do and the previous section is just for requesting features, which kilt will have to do...
I think it needs a bit more thought on how to actually structure these sections.

Dudleyneedham replied:
Updated

Dudleyneedham committed:
feat: altering some details about the contribution guide

tjwelde commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,58 @@
+
+## Treasury Proposals
+
+A treasury proposal is a request to spend funds within the treasure pool. Usually, the proposals have altered or enhanced KILT to better the protocol in a meaningful way. The proposal should begin with a proposal with a clear title and a written outline of the idea and a discussion about implementation or deliverables as outlined above for feature requests.
```
```suggestion
A treasury proposal is a request to spend funds from the treasure pool. Usually, the proposals would alter or enhance KILT to better the protocol in a meaningful way. The proposal should begin with a clear title and a written outline of the idea and a discussion about implementation or deliverables as outlined above for feature requests.
```

Dudleyneedham committed:
Update docs/development/01-contribution.md

Co-authored-by: Timo Welde <tjwelde@gmail.com>

Dudleyneedham committed:
feat: updating wording

Dudleyneedham committed:
feat: updating the wording from comms team

weichweich commented on docs/development/00-introduction.md:
```diff
@@ -0,0 +1,8 @@
+title: Development
+---
+
+KILT protocol is an open-source project built upon the [Substrate framework](https://docs.substrate.io/). The guide is to help builders and contributors find the resources needed to take action and work within the guidance of the active developers.
```
One sentence per line please. Better to review and better for git diffs.

```suggestion
KILT protocol is an open-source project built upon the [Substrate framework](https://docs.substrate.io/).
The guide is to help builders and contributors find the resources needed to take action and work within the guidance of the active developers.
```

Dudleyneedham committed:
Update docs/development/00-introduction.md

Co-authored-by: Albrecht <albrecht@kilt.io>

weichweich commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+As a decentralized network, KILT depends on the support of its community. There are many ways to contribute to KILT Protocol and the products and services built on it. 
+
+If you are interested in contributing but unsure how to begin, start in our [Clan KILT Discord](https://discord.gg/7uyfMXh6AT) channel. The developers active there, which include the team that originally developed KILT, and the wider KILT community can:
```
```suggestion
As a decentralized network, KILT depends on the support of its community.
There are many ways to contribute to KILT Protocol and the products and services built on it. 

If you are interested in contributing but unsure how to begin, start in our [Clan KILT Discord](https://discord.gg/7uyfMXh6AT) channel.
The developers active there, which include the team that originally developed KILT, and the wider KILT community can:
```

weichweich commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+## Treasury Proposals
+
+A Treasury proposal is a request to receive funds from the Treasury pool. The proposal should begin with a clear title, a written outline of the idea, and a discussion about implementation or deliverables as outlined above for feature requests. The proposal should be for something that changes or adds to KILT in a meaningful way. 
```
```suggestion
A Treasury proposal is a request to receive funds from the Treasury pool.
The proposal should begin with a clear title, a written outline of the idea, and a discussion about implementation or deliverables as outlined above for feature requests.
The proposal should be for something that changes or adds to KILT in a meaningful way. 
```

wischli replied:
I am missing an important piece of information which states that proposals are funding completed contributions and that one should rather go for multiple ones instead of a big proposal. I believe the best place for this statement is in this paragraph, so I added it to my suggestion below.

```suggestion
... adds value to KILT in a meaningful way. 
In general, a Treasury proposal spent occurs after completing all outlined deliverables and not before.
Thus, it is recommended to rather open multiple subsequent milestone-based proposals instead of a large one to fund contributions.
```

Dudleyneedham replied:
Yes, I like the addition. I will add change.

Dudleyneedham replied:
```suggestion
A Treasury proposal is a request to receive funds from the Treasury pool. 
The proposal should begin with a clear title, a written outline of the idea, and a discussion about implementation or deliverables as outlined above for feature requests. 
The proposal should be for something that changes or adds value to KILT in a meaningful way. 
In general, a Treasury proposal spent occurs after completing all outlined deliverables and not before.
Thus, it is recommended to rather open multiple subsequent milestone-based proposals instead of a large one to fund contributions.
```

weichweich commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+- Explain any deliverables that have already been achieved
+- Outline what needs to be done for the proposal to be completed
+
+The proposal should then be discussed with  the community (including, for example, the KILT Technical Committee, governance, or relevant project team) using community channels such as [Discord](https://discord.gg/7uyfMXh6AT). If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance. See the [guide to creating a proposal](https://www.kilt.io/tr ...
```
```suggestion
The proposal should then be discussed with  the community (including, for example, the KILT Technical Committee, governance, or relevant project team) using community channels such as [Discord](https://discord.gg/7uyfMXh6AT).
If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance.
See the [guide to creating a proposal](https://www.kilt.io/treasury/create-a-proposal/).
```

wischli commented on docs/development/00-introduction.md:
```diff
@@ -0,0 +1,9 @@
+KILT protocol is an open-source project built upon the [Substrate framework](https://docs.substrate.io/).
+The guide is to help builders and contributors find the resources needed to take action and work within the guidance of the active developers.
+
+- Do you want to start contributing? Check out our [contribution guide](./01-contribution.md) to get started and helping out!
```
The `helping out` feels off to me. Did Majella review this text already?

```suggestion
- Do you want to start contributing? Check out our [contribution guide](./01-contribution.md) to get started!
```

Dudleyneedham replied:
Yes, this is encouraging to participants.

weichweich commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+It is helpful to check if a report has already been filed in the related project. Search the issues board for possible phrases that match the description of the bug. It's possible you may not find an issue, but it's better to file a duplicated bug than not report one.
+
+Once you begin reporting the bug, write a descriptive title so that if others find the same issue they can either add to your findings or know that the bug has already been reported. A bug report should be as detailed as possible, including steps to reproduce, screenshots, error reports, or code snippets. The more details you provide, the easier it is to fix the issue.
```
```suggestion
We try our best, but bugs are an everyday reality with all software and are bound to happen.
We can't fix bugs we don't notice, so your input if you spot something gives us the best possibility to keep the project running smoothly and securely.

If you are unsure that a bug is a bug, it is best to open an issue and report it anyway.
The active developers will evaluate it and help to figure out the issue.

It is helpful to check if a report has already been filed in the related project.
Search the issues board for possible phrases that match the description of the bug.
It's possible you may not find an issue, but it's better to file a duplicated bug than not report one.

Once you begin reporting the bug, write a descriptive title so that if others find the same issue they can either add to your findings or know that the bug has already been reported.
A bug report should be as detailed as possible, including steps to reproduce, screenshots, error reports, or code snippets.
The more details you provide, the easier it is to fix the issue.
```

wischli replied:
```suggestion
Once you begin reporting the bug, write a descriptive title. Thus, if others find the same issue, they can either add to your findings or know that the bug has already been reported. A bug report should be as detailed as possible, including steps to reproduce, screenshots, error reports, or code snippets. The more details you provide, the easier it is to fix the issue.
```

weichweich commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+Pull Requests (PR) are an integral part of contributions to change KILT. GitHub itself has some [excellent documentation](https://help.github.com/articles/about-pull-requests/) on using the Pull Request feature. KILT uses the "[fork and pull](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models)" model described here, whe ...
+
+Before starting a PR, it’s best to contact other active developers and discuss the proposed changes. Open an issue or directly contact some of the devs on [Discord](https://discord.gg/7uyfMXh6AT) to begin discussing and making the proposal. Once approved, contributors can open a PR for review. The PR will be reviewed and, if accepted, merged into the corresponding repository.
```
```suggestion
Pull Requests (PR) are an integral part of contributions to change KILT.
GitHub itself has some [excellent documentation](https://help.github.com/articles/about-pull-requests/) on using the Pull Request feature.
KILT uses the "[fork and pull](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models)" model described here, where contributors push changes to their personal fork and create pull requests to bring those changes into the source repository.

Before starting a PR, it’s best to contact other active developers and discuss the proposed changes.
Open an issue or directly contact some of the devs on [Discord](https://discord.gg/7uyfMXh6AT) to begin discussing and making the proposal.
Once approved, contributors can open a PR for review.
The PR will be reviewed and, if accepted, merged into the corresponding repository.
```

weichweich committed:
Redo overview (#67)


Co-authored-by: William Freudenberger <w.freude@icloud.com>

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+As a decentralized network, KILT depends on the support of its community. There are many ways to contribute to KILT Protocol and the products and services built on it. 
+
+If you are interested in contributing but unsure how to begin, start in our [Clan KILT Discord](https://discord.gg/7uyfMXh6AT) channel. The developers active there, which include the team that originally developed KILT, and the wider KILT community can:
```
New sentence, new line.

```suggestion
If you are interested in contributing but unsure how to begin, we recommend to start in our [Clan KILT Discord](https://discord.gg/7uyfMXh6AT). The developers there, which include the team that originally developed KILT as well as the wider KILT community, can:
```

Dudleyneedham replied:
We have to use active developers when we refer the KILT community.

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+- Provide feedback on proposals or ideas
+- Discuss possible use-cases or feature requests
+- Make suggestions for non-technical contributions, including events, writing, or business models
+- Answer questions about the protocol, services and products.
```
I would add either periods or commas to each bullet point:

```suggestion
- Provide feedback on proposals or ideas.
- Discuss possible use-cases or feature requests.
- Make suggestions for non-technical contributions, including events, writing, or business models.
- Answer questions about the protocol, services and products.
```

Dudleyneedham replied:
Majella said no to this style and said at the end of all bullet points a fullstop is fine.

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+## Feature Requests 
+
+A feature request may be used to change KILT and its services by adding, removing or altering features within the scope of KILT Protocol.
```
```suggestion
A feature request may be used to change the KILT Protocol and its services by adding, removing or altering features within the scope the protocol.
```

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+A feature request is a meaningful way to contribute from anyone following the guidelines below:
+
+ - Begin a discussion with the community to ensure that they agree that it adds to the KILT protocol in a meaningful way, supporting its goals. 
```
```suggestion
 - Begin a discussion with the community to ensure that they agree that it adds to the KILT Protocol in a meaningful way, supporting its goals. 
```

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+ - Begin a discussion with the community to ensure that they agree that it adds to the KILT protocol in a meaningful way, supporting its goals. 
+ - Open an issue on the corresponding repository
+ - Give your request a clear title
+ - Provide a written outline of the feature request for discussion
```
Again, would add periods to align with the other bullet point
```suggestion
 - Open an issue on the corresponding repository.
 - Give your request a clear title.
 - Provide a written outline of the feature request for discussion.
```

wischli replied:
@Dudleyneedham According to your statement above, you have to move the period to the last bullet point then.

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+ - Give your request a clear title
+ - Provide a written outline of the feature request for discussion
+
+After discussion, if the community agrees that the change should be implemented, the proposer may also make a Treasury proposal to support the work. Check out the treasury proposals section.
```
```suggestion
If the community agrees that the change should be implemented after discussion, the proposer may also make a Treasury proposal to support the work. Check out the [treasury proposals section](#treasury-proposals) below.
```

Dudleyneedham replied:
I like the linkage, but the sentence reads better with `After discussion`

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+- Explain any deliverables that have already been achieved
+- Outline what needs to be done for the proposal to be completed
+
+The proposal should then be discussed with  the community (including, for example, the KILT Technical Committee, governance, or relevant project team) using community channels such as [Discord](https://discord.gg/7uyfMXh6AT). If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance. See the [guide to creating a proposal](https://www.kilt.io/tr ...
```
Adding to Albi's request for new lines: I would use KILT Council instead of the technical committee and including `governance` to `community` does not make sense to me. Thus, I would rephrase it as `community and council` as the council decides on Treasury proposals, not the Technical Committee and not the normal governance (for now).
```suggestion
The proposal should then be discussed with the community and the KILT council using community channels such as [Discord](https://discord.gg/7uyfMXh6AT) or [Polkassembly](https://kilt.polkassembly.network/discussions).
If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance.
Please read through the [guide on creating a proposal](https://www.kilt.io/treasury/create-a-proposal/).
```

Dudleyneedham replied:
I think it is good to include different aspects of the community. The technical committee is a role within the community and can take part in discussions. 

Though I will include the polkassembly part.

Dudleyneedham replied:
```suggestion
The proposal should then be discussed with  the community (including, for example, the KILT Technical Committee, governance, or relevant project team) using community channels such as [Discord](https://discord.gg/7uyfMXh6AT) or [Polkassembly](https://kilt.polkassembly.network/discussions).
If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance.
See the [guide to creating a proposal](https://www.kilt.io/treasury/create-a-proposal/).
```

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+
+- Explain any deliverables that have already been achieved
+- Outline what needs to be done for the proposal to be completed
```
It's perio'clock

```suggestion

- Explain any deliverables that have already been achieved.
- Outline what needs to be done for the proposal to be completed.
```

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+The proposal should then be discussed with  the community (including, for example, the KILT Technical Committee, governance, or relevant project team) using community channels such as [Discord](https://discord.gg/7uyfMXh6AT). If the community is not in agreement with the proposal, it is unlikely that it would be approved by governance. See the [guide to creating a proposal](https://www.kilt.io/tr ...
+
+Governance approval for a Treasury spend will also  includes list of the deliverables as outlined in the proposal.
```
What do you want to say with this statement? Governance approval is a boolean action. Of course, it can only occur, if all proposed deliverables are completed.

Dudleyneedham replied:
Yes, but that is only obvious if you know about the governance. It is a new concept and the readers potentially will not know about this. Therefore, we must be explicit in our speech and not let the reader knows.

wischli replied:
IMO, Governance mainly refers to Democracy (e.g. the entire community can vote in a public referendum). In the case of a Treasury proposal, voices can be raised but eventually the Council votes on it.

wischli replied:
In any case, the language is slightly off, isn't it?
```suggestion
Governance approval for a Treasury spend will also include a list of the deliverables as outlined in the proposal.
```

Dudleyneedham replied:
We can get rid of this statement.  :)

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+## Bug Reports
+
+We try our best, but bugs are an everyday reality with all software and are bound to happen. We can't fix bugs we don't notice, so your input if you spot something gives us the best possibility to keep the project running smoothly and securely.
```
I would like to improve the wording of this paragraph. Of course, the sentences need to be split into new lines but then the changes are harder to follow.

```suggestion
We try our best, but bugs are an everyday reality in software and are bound to happen. We cannot fix bugs which we do not notice. Thus, to give us the best possibility to keep the project running smoothly and securely, we encourage our community to inform us about any strange behaviour that you notice. Any input is valuable, so please do not hesitate to raise questions.
```

Dudleyneedham replied:
It feels repetitive. The two sentences are similar and the next sentence reinforces the issue.

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+Pull Requests (PR) are an integral part of contributions to change KILT. GitHub itself has some [excellent documentation](https://help.github.com/articles/about-pull-requests/) on using the Pull Request feature. KILT uses the "[fork and pull](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models)" model described here, whe ...
+
+Before starting a PR, it’s best to contact other active developers and discuss the proposed changes. Open an issue or directly contact some of the devs on [Discord](https://discord.gg/7uyfMXh6AT) to begin discussing and making the proposal. Once approved, contributors can open a PR for review. The PR will be reviewed and, if accepted, merged into the corresponding repository.
```
```suggestion
Before starting a PR, it is best to contact other active developers and discuss the proposed changes. Open an issue or directly contact some of the developers on [Discord](https://discord.gg/7uyfMXh6AT) to begin discussing and making the proposal. Once approved, contributors can open a PR for review. The PR will be reviewed and, if accepted, merged into the corresponding repository.
```

wischli replied:
We could also add mentioning of tips but should check with Gustav beforehand whether the phrasing is legally fine. Also, we might need an entire section on tips which would be out of scope of this PR. If you want, I can open a follow-up PR to add a section about tips.
```
Depending on the added value and scope of the contribution, it might be eligible for receiving a tip from the KILT Treasury.
```

Dudleyneedham replied:
At the moment, I think we can leave the TIP out. Since we haven't had a chance for Gustav to check and its fine since we can include it later once we have more documentation relevant to the TIP otherwise it feels slapped in there.

wischli commented on docs/development/01-contribution.md:
```diff
@@ -0,0 +1,59 @@
+
+Before starting a PR, it’s best to contact other active developers and discuss the proposed changes. Open an issue or directly contact some of the devs on [Discord](https://discord.gg/7uyfMXh6AT) to begin discussing and making the proposal. Once approved, contributors can open a PR for review. The PR will be reviewed and, if accepted, merged into the corresponding repository.
+
+The following section is inspired by Rust langs [Bug Report](https://rustc-dev-guide.rust-lang.org/contributing.html) contribution guide.
```
```suggestion
The following section is inspired by the Rust Programming Language [Bug Report](https://rustc-dev-guide.rust-lang.org/contributing.html) contribution guide.
```

wischli reviewed (changes requested):
I have a few language/wording remarks, as well as some content additions. 

Also, please migrate all sentences to a new line. This makes reviewing and editing way easier!

weichweich committed:
feat: add whitepaper disclaimer (#69)


Co-authored-by: William Freudenberger <w.freude@icloud.com>

Dudleyneedham committed:
Apply suggestions from code review

Co-authored-by: Albrecht <albrecht@kilt.io>
Co-authored-by: William Freudenberger <w.freude@icloud.com>

tjwelde committed:
feat: new navigation (#75)

trusch committed:
feat: add 'What is Kilt' page (#74)

Co-authored-by: Skedley <48058984+Dudleyneedham@users.noreply.github.com>
Co-authored-by: William Freudenberger <w.freude@icloud.com>

wischli committed:
fix: className (#77)

* fix: className

* fix: remove file

tjwelde committed:
refactor: rename chain menu entry (#78)

wischli committed:
refactor: update and improve collator and delegator docs (#71)

* refactor: improve collator section

* style: add secondary clr, nested tab

* refactor: new lines

* fix: num of active collators on Spiritnet

* refactor: delegator section

* refactor: empty lines in info box

* feat: add more delegator diagrams

* style: support full width for more mermaid diagrams

* refactor: update chain apps images

* docs: final improvements before review

* fix: rm file

* fix: apply suggestions from code review

* fix: revert pages/index

* fix: typo

* refactor: BTE support hub

ntn-x2 committed:
docs: add DID section in concepts (#82)

* docs: add DID section in concepts

* chore: add Polkadot ecosystem

* chore: add warning note about the difference between accounts and DIDs

* docs: add lookup diagram

* chore: use Dotsama instead

* chore: remove outstanding ;

* docs: add drawio files for light and dark did lookup diagrams

* chore: Internet -> internet

* chore: L2 -> higher level

* chore: hence -> therefore

* chore: rephrase warning

ntn-x2 committed:
docs: adjust DID core feature section (#84)

* docs: update DID core feature section

* chore: make KILT uppercase

weichweich committed:
feat: dev setup (#81)

* feat: dev setup

* fix: let vs const

* update dev setup

* fix: sidebar link

* Apply suggestions from code review

Co-authored-by: William Freudenberger <w.freude@icloud.com>

* mention encoding differences

* shorter headline, minor change

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

* Update docs/develop/sdk/1-dev-setup.md

Co-authored-by: Antonio <antonio@kilt.io>

* kilt -> Kilt

* fix npm & yarn

* only yarn

* its

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

* keyring from kilt

* make ntn-x2 happy

* explanations for empty accounts

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

Co-authored-by: William Freudenberger <w.freude@icloud.com>
Co-authored-by: Antonio <antonio@kilt.io>

weichweich committed:
fix: unsafe external for docker (#86)

ntn-x2 commented:
@Dudleyneedham is this still relevant with the new docs structure? If so, can we do anything to bring this task to completion?

ntn-x2 committed:
fix: fullnode instructions (#87)

* Update fullnode.md

1. fix docker run command with -v/ --volumn parameter
2. you can run a peregrine node using built-in config files for 'chain=' parameter

* chore: remove Peregrine snippet + Docker mount

* chore: add Spiritnet/Peregrine tab and fix chain command

* fix: capitalization

* chore: add tabs also for binary

* fix: relay chain spec for peregrine commands

Co-authored-by: whgreate <624906158@qq.com>

weichweich committed:
feat: suggest using your own full node (#90)

fixes KILTprotocol/ticket/issues/1843

Dudleyneedham committed:
feat: slim down getting started (#80)

* feat: slim down getting started

* feat: getting started guide

* feat: adding the getting started to the sidebars

* feat: adding highlighted text for user

* feat: updating wording and styling

* feat: adding some excution

* feat: fixing minor styling issues

* feat: adding git comments and example code

* feat: adding the testing of the examples

* feat: updating the code examples with snippetblock much cleaner

* feat: updating some lets to let us

* feat: adding display of did to the example 5

* feat: updating the connect from get connect

* feat: updating the styling and code blocks

* feat: fixing the npm2yarn styling issue

* feat: adding the node fetch to package

* feat: updating the script to use axios and code examples

wischli committed:
fix: Mermaid darkmode (#76)

* refactor: improve collator section

* style: add secondary clr, nested tab

* refactor: new lines

* fix: num of active collators on Spiritnet

* refactor: delegator section

* refactor: empty lines in info box

* feat: add more delegator diagrams

* style: support full width for more mermaid diagrams

* refactor: update chain apps images

* docs: final improvements before review

* fix: rm file

* fix: className

* refactor: move custom css

* fix: mermaid darkmode

* fix: footer colors

* 🧹🐢

* remove old things

Co-authored-by: weichweich <albrecht@kilt.io>

weichweich committed:
feat(concept): distributed trust (#88)

* wip

* rewrite

* simplify

* more decentralized not wider

* no strict hierarchy

* rename distributed trust

* darkmode svgs

* Make it better 😌

* 🧹🐢

* for github?

* 🐢🧹

* feat: remove useBaseUrl

https://docusaurus.io/docs/docusaurus-core#useBaseUrl

* 🧹🐢

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>

* wording

* Examples

Co-authored-by: Antonio <antonio@kilt.io>

ntn-x2 committed:
chore: add explanation of the different deployments (#89)

* chore: add explanation of the different deployment and the services associated with each of them

* docs: add table with different services for Spiritnet and Peregrine

* chore: adjust title of new section

* chore: update table based on feedback

* fix links

Co-authored-by: weichweich <albrecht@kilt.io>

weichweich committed:
feat: add note about ctype index (#91)

weichweich committed:
feat: Startpage Update (#92)

* feat: rewrite

* placeholder

* dark mode

* Sporran!

* update text

* add show case

* align logos

* Intro, make logos smaller

* Single brrrrr

tjwelde committed:
feat: new menu / sidebar structure (#93)

* feat: new menu / sidebar structure

* feat: menus and sidebar config

* feat: remove sdk intro and restructure some docs

* fix: links

* feat: exchange develop with participate

* feat: move collating & delegating to staking section

* feat: rename 'about delegating'

* fix: link

* fix: links

ntn-x2 committed:
docs: remove 'Apps' section and add 'Specifications' section (#95)

* chore: remove apps section

* docs: add section about specifications and WIP notice

* chore: remove navbar and change page name

weichweich committed:
style: better layout for showcase (#94)

* wip

* fix: imgs and small viewport

* deps: add react-markdown

* fix: markdown child props render

* docs: add small intro to apps

* fix: w3n.id logo alt

* fix: logo sizes

Co-authored-by: wischli <w.freude@icloud.com>

ntn-x2 committed:
docs: correct structure (#98)

* refactoring

* chore: implement structure details in the HackMD document

* fix: broken links

ntn-x2 committed:
docs: Web3 names & account linking (#99)

* fix: broken link in showcase

* chore: rename DID section to full name

* chore: split up web3 names concepts into their own section

* feat: add account linking with no SDK snippet

* docs: add cookbook for account -> Web3 name lookup

* feat: add section and code about public credential retrieval for web3 name

* chore: lint

* fix: wait for the run_core_features script to run to the end

* fix: test script

* docs: add web3 names and account linking docs

* feat: add snippet for deposit claiming for account <-> DID link

* fix: code snippet

* chore: re-order sections for Web3 name and account linking

* chore: remove intro section in web3 name claiming cookbook

* chore: remove intro section in account linking cookbook

* chore: use italics for emphasis

* chore: fix typo

* fix: CP error in linking snippet

* lint: add max-len rule and fix errors

* chore: refer -> refers

* chore: remove all signs of reSign option

ntn-x2 committed:
docs: DID cookbook sections (#101)

* docs: add cookbook about light and full DID creation

* docs: add DID cookbook sections

* fix: broken links

* chore: complete DID cookbook

* fix: credential publishing example snippet

* fix: core features test for light DID migration

* chore: linting

* revert: changes to test script (for now)

* test: add reSign true to see if it fails

* chore: remove empty line

* chore: reword light DID section

* fix: split new line where needed

* chore: on -> to

* chore: rephrasing full DID batching cookbook

wischli committed:
refactor: restructure and improve Staking documentation (#72)

* refactor: improve collator section

* style: add secondary clr, nested tab

* refactor: move collator sections to files

* fix: cleanup merge

* feat: add sidebar to collator sub section

* feat: restructure staking sections

* chore: add vscode to gitignore

* feat: add collator overview

* feat: add delegator overview

* refactor: rn files, add stakeboard disclaimers

* feat: normalize staking tx disclaimer

* feat: add missing top candidates screens

* style: improve mermaid centering

* fix: add missing disclaimer

* chore: final polish before review

* fix: mention peregrine spec, normalize full node + node setup

* fix: missing sign and submit improvs

* fix: incorrect merge

* Apply suggestions from code review

Co-authored-by: Antonio <antonio@kilt.io>
Co-authored-by: weichweich <albrecht@kilt.io>
Co-authored-by: Tino Rusch <tino@kilt.io>
Co-authored-by: Antonio <antonio@kilt.io>

ntn-x2 committed:
chore: major refactoring of test cases (#103)

* chore: major refactoring of test cases

* fix: small fixes

* chore: factor out resolution logic

* test: parallelize some test cases

* fix: special case handling for getting started flow

* fix: getting started snippets

* fix: links

* chore: change IS_IN_BLOCK to IS_FINALIZED for core features test

* chore: rename sidebar items

* fix: close connection after getting started regardless of completion status

* fix: broken links

* feat: make SnippetBlock work on functions

* feat: move quickstart guide into separate files and extract with new SnippetBlock

* feat: remove unneeded SnippetBlock imports

* chore: refactor test cases

* chore: style

Co-authored-by: Timo Welde <tjwelde@gmail.com>

wischli committed:
refactor: split up dev chain guide (#106)

tjwelde committed:
refactor(getting-started): no any & add FIXME comment (#105)

ggera committed:
Test docs agains new sdk rc (#83)

* init

* fix: resolve PR comments

* Removes lint and duplicate yarn installs

* Minor

* Adds high label to the issue

tjwelde committed:
feat: move credential docs into concept section (#85)

* feat: move credential docs into concept section

* feat: revise CType section

* feat: missing information for ctype

* feat: revise claiming section and remove code examples

* feat: revise the terms section

* refactor: rename terms docs

* feat: add a credential overview page

* fix: wait until crypto is ready

* fix: broken link

* fix: language in quote

* feat: highlight some words in the overview

* refactor: move terms to advanced

* chore: partially revert 16deedfa9887ace6081b5b7b48e078c171342f34

* feat: correct structure

* fix: PR review

* fix: PR feedback

* feat: move nested ctypes into advanced

* feat: reword claiming intro

* feat: reword attestation section

* refactor: rename to attestation

* feat: small changes in verification

* feat: replace ctype intro with text from claiming

* feat: reword nested ctypes intro

* feat: e.g. -> i.e.

* fix: broken link

* feat: pr review

* feat: restructure concept section

* feat: some improvements

* feat: add overview diagram

* feat: correct hashing properties

* feat: Req4Att example and describe it instead

* feat: link to selective disclosure section

* feat: remove ctype metadata example

* feat: restore cookbook files to develop

* feat: improvements in the attesation section

* feat: improvements in the terms and quote section

* feat: removal of nested ctype example

* feat: added section about requesting a credential

* feat: small improvements

ntn-x2 committed:
docs: add cookbook snippets for attestations and verifications (#104)

* fix: refactor test cases and add credential cookbook snippets

* chore: all Kilt imports from the package entrtypoint

* feat: claiming cookbook complete

* chore: style

* chore: rewording

* chore: small fixes and lints

* chore: one sentence one line

* fix: more ctype -> CType

rflechtner committed:
docs: messaging & replay protection (#100)

Adds a concepts section on messaging, and an advanced concepts section and cookbook examples on replay protection.

Co-authored-by: Antonio <antonio@kilt.io>

ggera committed:
Adds logic to update docs only when official sdk is released (#109)

* Adds logic to update docs during official sdk release only

* Minor

* removes beta check

trusch committed:
feat: add treasury proposal page from the homepage to the docs (#96)

* feat: add treasury proposal page from the homepage to the docs

Co-authored-by: William Freudenberger <w.freude@icloud.com>
Co-authored-by: Antonio <antonio@kilt.io>

trusch committed:
feat: add how-to-integrate docs (#110)

* feat: add how-to-integrate docs

* some formatting

* fix bug in example code

* add vitejs link

* minor fix to vitejs example

* Update docs/develop/01_sdk/04_How to integrate/01_nodejs.md

wischli committed:
feat: add tips section (#102)

* feat: add tips section

* docs: improve tips section

wischli committed:
Merge remote-tracking branch 'origin/develop' into dn-add-contribution-guide

ntn-x2 committed:
chore: editorial review (#111)

* review: US english

* review: everything web3name

* review: headings style

* fix: broken links

* review: uniform headings

* review: fix one sentence one line rule

* review: add WIP disclamer to WIP sections

* fix: SnippetBlock supports multi-line function signatures

* review: most of code snippets are either testable or have a TODO to become testable

* fix: ignore IPFS error on web3name tests

* wip: refactoring all docs

* review: finish the concepts

* review: getting-started

* review: SDK chain setup

* review: complete develop

* review: complete

* review: check heading capitalization

* fix: use node 16 in CI

* fix: broken links

* fix: broken links pt. 2

* wip: test remove cookbook links

* wip: remove instead of commenting broken links

* wip: remove leftover links instead of commenting them

* fix: make paths case sensitive

* Revert "wip: remove leftover links instead of commenting them"

This reverts commit 42da546057d4b433a26562cea979f0c9b6516a2e.

* Revert "wip: remove instead of commenting broken links"

This reverts commit 79fe322df14a66ef875283dc38e63210fa3ddd45.

* Revert "wip: test remove cookbook links"

This reverts commit 10e176f979b7336ad03ae3b28e38e1022479fd84.

* fix: broken links

* fix: remove unneeded yarn installation

* fix: revert CI Node version to 14.x

* Revert "fix: revert CI Node version to 14.x"

This reverts commit ccf74e9b12b579495c22698341ed47d8d8db2585.

* fix: revert yarn.lock changes

* wip: address William's PR comments

* review: address William's PR comments pt.2

* fix: capitalize remaining Credential

* fix: capitalize remaining Claimer

* fix: capitalize remaining Attester

* fix: capitalize remaining Verifier

* fix: case sensitivity for imports

* review: Tino's comments

* review: Tino's comments pt.2

* review: make KILT integrate snippets testable

* fix: style issue not caught locally

* chore: better naming of CI jobs

Co-authored-by: wischli <w.freude@icloud.com>

wischli committed:
Merge remote-tracking branch 'origin/develop' into dn-add-contribution-guide

wischli committed:
refactor: move contribution guideline

wischli commented:
Closed in favor of #113
